crates/tracing_macros/src/autotrace.rs

```diff
@@ -0,0 +1,97 @@
+use proc_macro::TokenStream;
+use quote::quote;
+use syn::{Attribute, ImplItemMethod, ItemFn};
+
+const DEBUG_OUTPUT: bool = false;
+
+#[derive(Default)]
+pub(crate) struct Autotrace {}
+
+impl Autotrace {
+    fn is_no_autotrace(&self, i: &[Attribute]) -> bool {
+        i.iter().any(|ref a| {
+            if a.path.segments.len() == 1 {
+                let ident = &a.path.segments.iter().next().unwrap().ident;
+                ident == "autotrace" || ident == "no_autotrace"
```

Comment:
I'm confused by this logic. Why are both conditions valid for no autotrace?

Reply:
Yeah I find this confusing too, but it works. I picked this trick up from flamer. It appears that when this Fold happens directly on an item that has the autotrace attr, the attr is not included (I guess because it was the attr that caused the Fold to happen in the first place?), but when the item is encountered as part of a larger AST, this prevents the fold from occurring during that pass thru, and lets the attr itself do the work.

For instance if you had `#[autotrace]` on both a `mod` and a `fn` in that module, when folding the `mod`, the fn would get folded twice, so this tells the mod fold to skip it because it's going to get handled by the attr directly on the fn.

Reply:
I thought it was something along those lines. Thanks for the explanation!